test/tools/runner/config.ts

```diff
@@ -427,6 +435,47 @@ export class TestConfiguration {
   makeAtlasTestConfiguration(): AtlasTestConfiguration {
     return new AtlasTestConfiguration(this.uri, this.context);
   }
+
+  loggingEnabled = false;
+  logs = [];
+  /**
+   * Known flaky tests that we want to turn on logging for
+   * so that we can get a better idea of what is failing when it fails
+   */
+  testsToEnableLogging = [
+    'Client Side Encryption (Unified) namedKMS-rewrapManyDataKey rewrap to azure:name1',
+    'Server Discovery and Monitoring Prose Tests Connection Pool Management ensure monitors properly create and unpause connection pools when they discover servers',
+    'CSOT spec tests legacy timeouts behave correctly for retryable operations operation succeeds after one socket timeout - aggregate on collection'
+  ];
+
+  setupLogging(options) {
+    this.logs = [];
+    const write = log => this.logs.push(log);
+    options.mongodbLogPath = { write };
+    options.mongodbLogComponentSeverities = { default: 'trace' };
+    options.mongodbLogMaxDocumentLength = 300;
+    return options;
+  }
+
+  beforeEachLogging(ctx: Context) {
+    this.loggingEnabled = this.testsToEnableLogging.includes(ctx.currentTest.fullTitle());
+  }
+
+  afterEachLogging(ctx: Context) {
+    if (this.loggingEnabled && ctx.currentTest.state === 'failed') {
+      for (const log of this.logs) {
+        const logLine = util.inspect(log, {
+          compact: true,
+          breakLength: Infinity,
+          colors: true,
+          depth: 1000
+        });
+        console.error(logLine);
```

Comment:
Everything that comes out of the logger is valid JSON.  Can we use `JSON.stringify` instead, so that the output is parsable with `jq` or a Node program?

Reply:
That isn't true, some values won't stringify correctly like the [serverConnectionId](https://parsley.mongodb.com/evergreen/mongo_node_driver_next_rhel80_large_Node20_test_8.0_replica_set_patch_9111f98cadcb594ab7c4fd3161f2f6e42105f917_67eecac9e10ab70007680e9a_25_04_03_17_52_20/0/task?bookmarks=0,1973,9706&shareLine=1973)

Reply:
I'd still like some mechanism to interact with the logs programmatically, that's the first thing I do whenever I look at logs.  Could we stringify with a custom reviver?

Reply:
switched to json and added a replacer